colcon_ros_cargo/package_identification/ament_cargo.py

```diff
@@ -22,17 +23,25 @@ def __init__(self):  # noqa: D107
             '^1.0')
 
     def identify(self, metadata):  # noqa: D102
-        if metadata.type is not None and metadata.type != 'ament_cargo':
-            return
+        from catkin_pkg.package import parse_package
 
-        cargo_toml = metadata.path / 'Cargo.toml'
-        if not cargo_toml.is_file():
+        if metadata.type is not None and metadata.type != 'ament_cargo':
             return
 
         package_xml = metadata.path / 'package.xml'
         if not package_xml.is_file():
             return
 
+        pkg_desc = parse_package(package_xml)
+
+        if pkg_desc.get_build_type() != 'ament_cargo':
+            return
+
+        cargo_toml = metadata.path / 'Cargo.toml'
+        if not cargo_toml.is_file():
+            logger.warn('Got build type ament_cargo but could not find "Cargo.toml"')
+            raise RuntimeError('Got build type ament_cargo but could not find "Cargo.toml"')
```

Comment:
I think just doing one of the two operations should be enough, rather than raising an exception and log a warning with the same message. I suppose just a warning + return would be enough?

This is an example of the current output which is a bit messy:

```
[0.187s] WARNING:colcon.colcon_core.package_identification:Got build type ament_cargo but could not find "Cargo.toml"
[0.188s] ERROR:colcon.colcon_core.package_identification:Exception in package identification extension 'ament_cargo' in 'src/rust-sample-package': Got build type ament_cargo but could not find "Cargo.toml"
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/colcon_core/package_identification/__init__.py", line 144, in _identify
    retval = extension.identify(_reused_descriptor_instance)
  File "/home/lucadv/colcon_ws/install/colcon-ros-cargo/lib/python3.10/site-packages/colcon_ros_cargo/package_identification/ament_cargo.py", line 43, in identify
    raise RuntimeError('Got build type ament_cargo but could not find "Cargo.toml"')
RuntimeError: Got build type ament_cargo but could not find "Cargo.toml"
```

Reply:
9439dd1188725caf6fb65a29a04eb4c5aa4e2551